src/settings.py

```diff
@@ -161,222 +71,122 @@ def settings_set(key, value):
-                    setting=key,
-                    received_type=type(value).__name__,
-                    expected_type=key_type,
-                )
-    elif key_type == "string":
-        if not isinstance(value, str):
-            raise YunohostValidationError(
-                "global_settings_bad_type_for_setting",
-                setting=key,
-                received_type=type(value).__name__,
-                expected_type=key_type,
-            )
-    elif key_type == "enum":
-        if value not in settings[key]["choices"]:
-            raise YunohostValidationError(
-                "global_settings_bad_choice_for_enum",
-                setting=key,
-                choice=str(value),
-                available_choices=", ".join(settings[key]["choices"]),
-            )
-    else:
-        raise YunohostValidationError(
-            "global_settings_unknown_type", setting=key, unknown_type=key_type
-        )
-
-    old_value = settings[key].get("value")
-    settings[key]["value"] = value
-    _save_settings(settings)
-
-    try:
-        trigger_post_change_hook(key, old_value, value)
-    except Exception as e:
-        logger.error(f"Post-change hook for setting {key} failed : {e}")
-        raise
-
-
-def settings_reset(key):
+@is_unit_operation()
+def settings_reset(operation_logger, key):
     """
     Set an entry value to its default one
 
     Keyword argument:
         key -- Settings key
 
     """
-    settings = _get_settings()
-
-    if key not in settings:
-        raise YunohostValidationError(
-            "global_settings_key_doesnt_exists", settings_key=key
-        )
 
-    settings[key]["value"] = settings[key]["default"]
-    _save_settings(settings)
+    settings = SettingsConfigPanel()
+    key = translate_legacy_settings_to_configpanel_settings(key)
+    return settings.reset(key, operation_logger=operation_logger)
 
 
-def settings_reset_all():
+@is_unit_operation()
+def settings_reset_all(operation_logger):
     """
     Reset all settings to their default value
 
     Keyword argument:
         yes -- Yes I'm sure I want to do that
 
     """
-    settings = _get_settings()
+    settings = SettingsConfigPanel()
+    return settings.reset(operation_logger=operation_logger)
 
-    # For now on, we backup the previous settings in case of but we don't have
-    # any mecanism to take advantage of those backups. It could be a nice
-    # addition but we'll see if this is a common need.
-    # Another solution would be to use etckeeper and integrate those
-    # modification inside of it and take advantage of its git history
-    old_settings_backup_path = (
-        SETTINGS_PATH_OTHER_LOCATION % datetime.utcnow().strftime("%F_%X")
-    )
-    _save_settings(settings, location=old_settings_backup_path)
 
-    for value in settings.values():
-        value["value"] = value["default"]
+class SettingsConfigPanel(ConfigPanel):
+    entity_type = "settings"
```

Comment:
```suggestion
    entity_type = "global"
```

Should be less confusing ? Since usually we talk about "domain settings" or "app settings" or "global settings", and entity_type is already "domain" and "app" for the other two ?